util/encoding/decimal.go

```diff
@@ -356,24 +394,23 @@ var digitsLookupTable [tableSize + 1]tableVal
 
 type tableVal struct {
 	digits int
-	border *big.Int
+	border big.Int
 }
 
 func init() {
-	digitBi := new(big.Int)
-	var bigIntArr [tableSize]big.Int
+	curVal := big.NewInt(1)
+	curExp := new(big.Int)
 	for i := 1; i <= tableSize; i++ {
-		val := int(1 << uint(i-1))
-		digits := 1
-		for ; val > 10; val /= 10 {
-			digits++
+		if i > 1 {
+			curVal.Lsh(curVal, 1)
 		}
 
-		digitBi.SetInt64(int64(digits))
-		digitsLookupTable[i] = tableVal{
-			digits: digits,
-			border: bigIntArr[i-1].Exp(bigInt10, digitBi, nil),
-		}
+		elem := &digitsLookupTable[i]
+		elem.digits = len(curVal.String())
+
+		elem.border.SetInt64(10)
```

Comment:
Yeah I could get rid of the `elem.border.Exp` for all elements in the power of 10 table and just `Set` directly. It would save one allocation in an `init` function though, so it's not super pressing. I'll see if I can clean it up once I address that power of ten caching.